chains/evm/contracts/offRamp/CCVAggregator.sol

```diff
@@ -289,66 +316,159 @@ contract CCVAggregator is ITypeAndVersion, Ownable2StepMsgSender {
   function _ensureCCVQuorumIsReached(
     uint64 sourceChainSelector,
     address receiver,
-    address tokenPool,
-    address[] calldata ccvs
-  ) internal view {
+    address[] calldata ccvs,
+    address[] memory tokenRequiredCCVs
+  ) internal view returns (address[] memory ccvsToQuery, uint256[] memory dataIndexes) {
     (address[] memory requiredCCV, address[] memory optionalCCVs, uint8 optionalThreshold) =
-      _getCCVsFromReceiverAndPool(sourceChainSelector, receiver, tokenPool);
+      _getCCVsFromReceiver(sourceChainSelector, receiver);
+    address[] memory laneMandatedCCVs = s_sourceChainConfigs[sourceChainSelector].laneMandatedCCVs;
+
+    ccvsToQuery = new address[](ccvs.length);
+    dataIndexes = new uint256[](ccvs.length);
```

Comment:
Haven't read this code thoroughly yet but is `ccvs.length` sufficient for these arrays? Cuz you have `ccvs` and `tokenRequiredCCVs`, couldn't they be entirely different and you'd end up having to append more to this array than `ccvs.length` items?

Reply:
ccvs is the passed in ccvs, so everything we have data for. You can never end up with more than that